corehq/apps/data_cleaning/forms/filters.py

```diff
@@ -100,11 +108,19 @@ def __init__(self, session, *args, **kwargs):
                     x_select2=json.dumps({
                         "placeholder": _("Select a Case Property"),
                     }),
+                    **({
+                        "@select2change": "propId = $event.detail; "
+                                          "dataType = casePropertyDetails[$event.detail].data_type; "
+                                          "isEditable = casePropertyDetails[$event.detail].is_editable;",
+                    })
```

Comment:
example of disabled data type field when a system property `prop_id` is selected:
<img width="370" alt="Screenshot 2025-03-11 at 4 16 53 PM" src="https://github.com/user-attachments/assets/6d3c25ac-39df-4d40-b073-490ba1eab0a6" />